bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/api/IConnectionStatusListener.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.metofficedatahub.internal.api;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * Implementations of this interface, allow access to a HttpClient which can be used
+ * for communication requests to LinkTap Gateways.
+ *
+ * @author David Goodyear - Initial contribution
+ */
+@NonNullByDefault
+public interface IConnectionStatusListener {
+    void processAuthenticationResult(final boolean authenticated);
+
+    void processCommunicationFailure(final @Nullable Throwable e);
+
+    void processConnected();
```

Comment:
Please add javadoc to all public interface methods

Reply:
Yes sorry I remembered that at the time, when I did them - that I must go back and do this from the previous MR / PR's. Apologies my bad. I've added to all the interfaces now.